analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParametersCorrectOrder.CSharp12.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public interface IInterface
+    {
+        static virtual void SomeMethod(int a, int b) { } // Secondary
+    }
+
+    public class SomeClass<T> where T : IInterface
+    {
+        public SomeClass()
+        {
+            int a = 1;
+            int b = 2;
+
+            T.SomeMethod(b, a); // Noncompliant
+        }
+    }
+}
+
+// https://github.com/SonarSource/sonar-dotnet/issues/8071
+namespace Repro_8071
+{
+    class BaseConstructor
+    {
+        class Base(int a, int b)
+        {
+            Base(int a, int b, string c) : this(b, a) { }  // FN: ctor params inverted with additional param after
+            Base(string c, int a, int b) : this(b, a) { }  // FN: ctor params inverted with additional param before
+        }
+
+        class ParamsFullyInverted(int a, int b) : Base(b, a);                                    // FN
+        class ParamsPartiallyInverted(int a, int b, int c) : Base(b, a);                         // FN
+        class ParamsFullyInvertedWithAdditionalParamAfter(int a, int b, string s) : Base(b, a);  // FN
+        class ParamsFullyInvertedWithAdditionalParamBefore(string s, int a, int b) : Base(b, a); // FN
+    }
+
+    class WithRecordStructs
+    {
+        void Basics(int a, int b, int c)
+        {
+            _ = new SomeRecord(b, a);           // Noncompliant
+        }
+
+        void WithPromotion(short a, short b)
+        {
+            _ = new SomeRecord(b, a);           // Noncompliant
+        }
+
+        void WithCasting(long a, long b)
+        {
+            _ = new SomeRecord((int)b, (int)a); // FN
+        }
+
+        record SomeRecord(int a, int b)
+        {
+            public SomeRecord(int a, int b, string c) : this(b, a) { } // FN
+            public SomeRecord(string c, int a, int b) : this(b, a) { } // FN
+        }
+    }
+
+    class WithRecords
+    {
+        void Basics(int a, int b, int c)
+        {
+            _ = new SomeRecordStruct(b, a);           // Noncompliant
+        }
+
+        void WithPromotion(short a, short b)
+        {
+            _ = new SomeRecordStruct(b, a);           // Noncompliant
+        }
+
+        void WithCasting(long a, long b)
+        {
+            _ = new SomeRecordStruct((int)b, (int)a); // FN
+        }
+
+        record struct SomeRecordStruct(int a, int b)
+        {
+            public SomeRecordStruct(int a, int b, string c) : this(b, a) { } // FN
+            public SomeRecordStruct(string c, int a, int b) : this(b, a) { } // FN
+        }
+    }
+}
+
+namespace Repro_8072
```

Comment:
Duplicate of the Repro_8072 in C#11, I would keep the one that is running the lowest C# version

Reply:
Since it's out-of-scope for this PR, it will be addressed in [a new, dedicated PR](https://github.com/SonarSource/sonar-dotnet/pull/8118).